packages/vitest/src/node/core.ts

```diff
@@ -351,18 +351,31 @@ export class Vitest {
   }
 
   async exit() {
-    const closePromise = this.close()
     let timeout: NodeJS.Timeout
+    let resolveClose: () => void
+    let resolveTimeout: () => void
+    const closePromise = new Promise((resolve, reject) => {
+      resolveClose = () => resolve(true)
+      this.close().then(resolve, reject)
+    })
+    // make sure all promises are resolved before exiting process
+    // https://github.com/nodejs/node/blob/master/src/node_file-inl.h#L160
+    const cleanup = () => {
+      clearTimeout(timeout)
+      resolveTimeout?.()
+      resolveClose?.()
+    }
     const timeoutPromise = new Promise((resolve, reject) => {
+      resolveTimeout = () => resolve(true)
```

Comment:
This could directly be the resolve function, no?
```suggestion
      resolveTimeout = resolve
```

Reply:
It can be, but it's type requires to pass something, so I just did it before 👀